sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+                                  + "Bulk insert api response==="
+                                  + bResponse.getId()
+                                  + " "
+                                  + bResponse.isFailed());
+                        }
+                      }
+                    }
+
+                    @Override
+                    public void afterBulk(
+                        long executionId, BulkRequest request, Throwable failure) {
+                      ProjectLogger.log(
+                          "ElasticSearchTcpImpl:bulkInsert: Bulk upload error block", failure);
+                    }
+                  })
+              .setBulkActions(10000)
+              .setConcurrentRequests(0)
+              .build();
+
+      for (Map<String, Object> map : dataList) {
+        map.put(JsonKey.IDENTIFIER, map.get(JsonKey.ID));
+        IndexRequest request =
+            new IndexRequest(index, _DOC, (String) map.get(JsonKey.IDENTIFIER)).source(map);
+        bulkProcessor.add(request);
+      }
+      // Flush any remaining requests
+      bulkProcessor.flush();
+
+      // Or close the bulkProcessor if you don't need it anymore
+      bulkProcessor.close();
+
+      // Refresh your indices
+      ConnectionManager.getClient().admin().indices().prepareRefresh().get();
+    } catch (Exception e) {
+      promise.failure(e);
+      promise.success(false);
+      ProjectLogger.log(e.getMessage(), e);
+    }
+    long elapsedTime = ElasticSearchHelper.calculateEndTime(startTime);
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:bulkInsert: method end  at =="
+            + System.currentTimeMillis()
+            + " for index "
+            + index
+            + " ,Total time elapsed = "
+            + elapsedTime,
+        LoggerEnum.PERF_LOG);
+    return promise.future();
+  }
+
+  /**
+   * This method will do the health check of elastic search.
+   *
+   * @return boolean
+   */
+  @Override
+  public Future<Boolean> healthCheck() {
+    Promise<Boolean> promise = Futures.promise();
+
+    boolean indexResponse = false;
+    try {
+      indexResponse =
+          ConnectionManager.getClient()
+              .admin()
+              .indices()
+              .exists(Requests.indicesExistsRequest(ProjectUtil.EsType.user.getTypeName()))
+              .get()
+              .isExists();
+    } catch (Exception e) {
+      ProjectLogger.log("ElasticSearchTcpImpl:healthCheck: error " + e.getMessage(), e);
+      promise.failure(e);
+    }
+    promise.success(indexResponse);
+    return promise.future();
+  }
+
+  /**
+   * Method to execute ES query with the limitation of size set to 0 Currently this is a rest call
+   *
+   * @param index ES indexName
+   * @param rawQuery actual query to be executed
+   * @return ES response for the query
+   */
+  @SuppressWarnings("unchecked")
+  @Override
+  public Response searchMetricsData(String index, String rawQuery) {
```

Comment:
As this method is nither related to TCP  nor high rest client, so use java 8 feature default implementation method and move it to interface only.
this will be applicable for both impl

Reply:
Done